### PR TITLE
feat(gen): generic_json_service

### PR DESCRIPTION
**Pull request by mattjoke**
- added generic_json_gen service
- this will generate structure according to the documentation
- removes empty structures
- pushing this here to have the quickest review

### CODE REVIEW

**Comment by mattjoke** [comment]
Merge AFTER CESNET/perun#4153

File: gen/generic_json_gen [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+}
+if (!exists($requiredAttributes->{"group"}->{$A_G_UUID})) {
+	die("Attribute $A_G_UUID is required for this service");
+}
+if (!exists($requiredAttributes->{"resource"}->{$A_R_UUID})) {
+	die("Attribute $A_R_UUID is required for this service");
+}
+if (!exists($requiredAttributes->{"user"}->{$A_U_UUID})) {
+	die("Attribute $A_U_UUID is required for this service");
+}
```
[the remark this answers is not in the document]
Comment (mattjoke) [reply gen/generic_json_gen:64]: Those are not loaded from the attribute but directly from the object. Nevertheless, I found a bug, so this part of the code will probably get fixed.

File: gen/generic_json_gen [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+	# Lookup update
+	$lookup->{uuids}->{$groupUUID} = $gropuId;
+	$lookup->{ids}->{group}->{$gropuId} = $groupUUID;
+
+	# Load Group attributes (undef if not present)
+	my %group_attributes;
+	for my $urn (keys %{$requiredAttributes->{group}}) {
+		my $value = $data->getGroupAttributeValue(group => $gropuId, attrName => $urn);
+		$group_attributes{$urn} = $value;
+	}
```
Comment (xflord) [inline gen/generic_json_gen:190]: pretty sure we can use `$data->getAllGroupAttributes`, which should return all the required attributes I believe, for this (and all other cases), which should speed things up. Checkout `denbi_portal_compute_center` gen script

**Comment by zlamalp** [comment]
This code has been merged in source repository. Closing this PR.

**Pull request closed without merging**